opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```diff
@@ -105,29 +111,14 @@ public void init(FilterConfig filterConfig) throws ServletException {
         this.filterConfig = filterConfig;
 
         // init if the tracer is null
-        if (tracer == null) {
-            ServletContext servletContext = filterConfig.getServletContext();
-
-            Object contextAttribute = servletContext.getAttribute(TRACER);
-            if (contextAttribute == null) {
-                log.severe("Tracer was not found in `ServletContext.getAttribute(TRACER)`, skipping tracing filter");
-                this.skipFilter = true;
-                return;
-            }
-            if (!(contextAttribute instanceof Tracer)) {
-                log.severe("Tracer from `ServletContext.getAttribute(TRACER)`, is not an instance of " +
-                        "io.opentracing.Tracer, skipping tracing filter");
-                this.skipFilter = true;
-                return;
-            }
-            this.tracer = (Tracer)contextAttribute;
-
-            this.spanDecorators = (List<SpanDecorator>)servletContext.getAttribute(SPAN_DECORATORS);
-            if (this.spanDecorators == null) {
-                this.spanDecorators = Arrays.asList(SpanDecorator.STANDARD_TAGS);
+        if (spanDecorators == null) {
+            spanDecorators = decoratorSPI();
+            if (spanDecorators.isEmpty()) {
+                this.spanDecorators = Arrays.<ServletFilterSpanDecorator>asList(new ServletFilterSpanDecorator.StandardTags());
```

Comment:
Why did you replace from `Collections#singleton()` to `Arrays#asList()` ? You have only one item and your target type is a collection, so, it's more appropriate to use a `Collections#singletonList` :)